packages/transaction-controller/src/TransactionController.ts

```diff
@@ -798,23 +797,43 @@ export class TransactionController extends BaseController<
             value: '0x0',
           };
 
-    const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
+    const unsignedEthTx = this.prepareUnsignedEthTx(newTxParams);
 
     const signedTx = await this.sign(
       unsignedEthTx,
       transactionMeta.txParams.from,
     );
-    await this.updateTransactionMetaRSV(transactionMeta, signedTx);
     const rawTx = bufferToHex(signedTx.serialize());
-    await query(this.ethQuery, 'sendRawTransaction', [rawTx]);
-    transactionMeta.estimatedBaseFee = estimatedBaseFee;
-    transactionMeta.status = TransactionStatus.cancelled;
+    const hash = await this.publishTransaction(rawTx);
+    const cancelTransactionMeta: TransactionMeta = {
```

Comment:
We've diverged a little from `speedUpTransaction` but I think that's a job for a future ticket where we can also remove all the duplication between them.